gpx.go

```diff
@@ -171,86 +161,55 @@ type MovingData struct {
 	MaxSpeed        float64
 }
 
-// SpeedsAndDistances represents speed and distance
-type SpeedsAndDistances struct {
-	Speed    float64
-	Distance float64
-}
-
-// LocationsResultPair represents a result from location query
-type LocationsResultPair struct {
-	SegmentNo int
-	PointNo   int
+// speedsAndDistances represents speed and distance
+type speedsAndDistances struct {
+	speed    float64
+	distance float64
 }
 
 /*==========================================================*/
 
-// Parse parses a GPX file and return a Gpx object.
-func Parse(gpxPath string) (*Gpx, error) {
-	gpxFile, err := os.Open(gpxPath)
+// Parse parses a GPX reader and return a Gpx object.
+func Parse(r io.Reader) (*Gpx, error) {
+	g := NewGpx()
+	d := xml.NewDecoder(r)
+	err := d.Decode(g)
 	if err != nil {
-		// fmt.Println("Error opening file: ", err)
-		return nil, err
+		return nil, fmt.Errorf("couldn't parse gpx data: %v", err)
 	}
-	defer gpxFile.Close()
-
-	b, err := ioutil.ReadAll(gpxFile)
+	return g, nil
+}
 
+// ParseFile reads a GPX file and parses it.
+func ParseFile(filepath string) (*Gpx, error) {
+	f, err := os.Open(filepath)
 	if err != nil {
-		// fmt.Println("Error reading file: ", err)
 		return nil, err
 	}
-	g := NewGpx()
-	xml.Unmarshal(b, &g)
-
-	return g, nil
+	return Parse(f)
 }
 
 /*==========================================================*/
 
-func getTime(timestr string) time.Time {
-	t, err := time.Parse(timelayout, timestr)
-	if err != nil {
-		return time.Time{}
-	}
-	return t
-}
-
 func toXML(n interface{}) []byte {
 	content, _ := xml.MarshalIndent(n, "", "	")
 	return content
 }
 
-func getMinimaMaximaStart() *Bounds {
-	return &Bounds{
-		MaxLat: -math.MaxFloat64,
-		MinLat: math.MaxFloat64,
-		MaxLon: -math.MaxFloat64,
-		MinLon: math.MaxFloat64,
-	}
+var maxBounds = Bounds{
```

Comment:
We can't have const structs in Go; only numbers, strings and booleans. I'm pretty sure I've seen unexported variables used as constants within a package in the std library but can't quite find it again...

I'm not even sure why I changed this from a function in the first place, that was sensible. I'll change it back.